### PR TITLE
feat: allow configuring multiple RPs for Kubernetes [DET-8711]

### PR DESCRIPTION
**Pull request by dzhu**
## Description

The RPs do not correspond to namespaces for the moment; this just sets up the ability to have multiple logical resource pools at all.

Some of the APIs return slightly incorrect values, since this is still pushing everything through a single pods actor, which is doing some summarization without being aware of the multiple RPs. I'm going to call that okay for this PR, since cluster behavior won't change without config changes, this is undocumented, and the plan is to do more work before the next release anyway.

Much of the new code in the RM is copied directly from the agent RM. I think it may be possible to fully unify them and move all agent/Kubernetes-specific code into the pools, seeing how both RMs are pretty much just managing the fleet of RP actors and doing some of mostly non-cluster-specific bookkeeping, but that's a task for the future.

## Test Plan

- [x] basic manual smoke tests
- [x] upstream CI tests

### CODE REVIEW

File: master/internal/rm/kubernetesrm/kubernetes_resource_manager.go
Comment (NicholasBlaskey) [inline master/internal/rm/kubernetesrm/kubernetes_resource_manager.go:261]: somewhat related think CLI has a bug of not passing ```args.resource_pool``` in ```det job ls``` 

https://github.com/determined-ai/determined/blob/b69bab1c1c59172963ec0f8eb2caba6cc6c24c1a/harness/determined/cli/job.py#L31

```
                    Arg(
                        "-p",
                        "--resource-pool",
                        type=str,
                        help="The target resource pool, if any.",
                    ),

```

File: master/internal/config/resource_manager_config.go
Comment (dzhu) [reply master/internal/config/resource_manager_config.go:118]: Updated Helm chart.

File: master/internal/config/resource_config.go
Comment (dzhu) [reply master/internal/config/resource_config.go:29]: Good catch, I knew it would change the name, but I didn't realize that would have any effects. It does break just clicking the fork button, but you can manually open the full config in the UI and change the resource pool. Think this warrants any workarounds in the code, or can we just call it a release note?

Comment (NicholasBlaskey) [reply master/internal/config/resource_config.go:29]: To fix it in the code we could make the default pool for k8s ```kubernetes``` which could be okay? 

Could we get away with a migration updating expconf where pool is ```kubernetes``` (would be pretty weird if someone had an agent pool named ```kubernetes```)?

Documenting could be fine too, but is a little unfortunate to cause an error with completely default master config upgrade

**Review by NicholasBlaskey — APPROVED** [review]
LGTM, I'll let you decide on what if anything should be done about forking

**Comment by dzhu** [comment]
Alright, since it is an easy user workaround, I'll plan to just make it a release note (not adding that here, since this PR by itself is not the full form of the visible feature).

**Pull request merged**
